compiler/src/dotty/tools/dotc/ast/DesugarEnums.scala

```diff
@@ -84,35 +92,55 @@ object DesugarEnums {
   private def valuesDot(name: PreName)(implicit src: SourceFile) =
     Select(Ident(nme.DOLLAR_VALUES), name.toTermName)
 
-  private def registerCall(using Context): Tree =
-    Apply(valuesDot("register"), This(EmptyTypeIdent) :: Nil)
+  private def ArrayLiteral(values: List[Tree], tpt: Tree)(using Context): Tree =
+    val clazzOf = TypeApply(ref(defn.Predef_classOf.termRef), tpt :: Nil)
+    val ctag    = Apply(TypeApply(ref(defn.ClassTagModule_apply.termRef), tpt :: Nil), clazzOf :: Nil)
+    val apply   = Select(ref(defn.ArrayModule.termRef), nme.apply)
+    Apply(Apply(TypeApply(apply, tpt :: Nil), values), ctag :: Nil)
```

Comment:
Can't you directly generate a `JavaSeqLiteral` here instead?

Reply:
yes thanks for the tip